window/ui/preload.js

```diff
@@ -0,0 +1,39 @@
+const { ipcRenderer, contextBridge } = require('electron')
+
+// We request that the main process sends us a channel we can use to
+// communicate with the backend
+let port
+ipcRenderer.send('request-channel')
+
+ipcRenderer.once('provide-ssb-channel', (event) => {
+  port = event.ports[0]
+  // ... register a handler to receive results ...
+  port.onmessage = (ev) => {
+    console.log('received from window/ssb:', ev.data)
+
+    if (_resolve) _resolve(ev.data)
+  }
+
+  let _resolve
+
+  contextBridge.exposeInMainWorld('ssb', {
```

Comment:
this makes it so `window.ssb` gives you access to these manually exposed methods!